### ISSUE TITLE
Clarify handling of `use_tranpoline` in `sdk-common` structs

### ISSUE THREAD

**Issue by ok300** [breez/breez-sdk-greenlight#1081]
There is one struct in particular that uses this behind a feature flag:

https://github.com/breez/breez-sdk-greenlight/blob/12980553a1ca1f6cd4d73e1128cb468276c578b6/libs/sdk-common/src/lnurl/specs/pay.rs#L128-L147

Question:
- Should `sdk-common` remove the `liquid` feature flag and instead only contain truly common structs and code?
  - If so, should there be a separate `LnUrlPayRequestWithTranpoline` (or similar) in the GL SDK repo?
- Alternatively, is the current approach ok and we should keep it?

Let's discuss below.

**Comment by JssDWt** [comment]
Is it possible to have `LnUrlPayRequest` in sdk greenlight and `LnUrlPayRequest` in sdk liquid, but they're different structs? That would be my preferred option.

**Comment by ok300** [comment]
IMO the feature flag in `sdk-common` is overkill. We should instead place only common code in `sdk-common`. What's implementation-specific should live in the implementation-specific repo (for example a new struct with `use_trampoline` in the GL SDK repo).

**Comment by ok300** [comment]
> Is it possible to have `LnUrlPayRequest` in sdk greenlight and `LnUrlPayRequest` in sdk liquid, but they're different structs? That would be my preferred option.

I don't think so, as `lnurl_pay` lives in `sdk-common` and it has to know which request struct to reference.

**Comment by JssDWt** [comment]
> I don't think so, as `lnurl_pay` lives in `sdk-common` and it has to know which request struct to reference.

Is it? I can't find it.

**Comment by JssDWt** [comment]
There might be another option here too.
`use_trampoline` could be a configuration-time setting instead. That would mean it applies to all payments. Then this problem would no longer exist, because the field would not be necessary on a specific payment. What do you think @roeierez?

**Comment by ok300** [comment]
You're right, `lnurl_pay` is not in `sdk-common` but instead implemented (slightly differently) in Liquid and GL SDK repos.

Then your idea could well work. The GL SDK repo `lnurl_pay` could reference its own request struct with `use_trampoline`, the Liquid one could reference the `sdk-common` request struct without `use_trampoline`.

**Comment by roeierez** [comment]
I think the whole problem is that the udl interface has leaked into the sdk-common. We should leave there only "business-logic" or business models.
So I don't see any issue in creating two different structures and two different udl definitions.

**Comment by JssDWt** [comment]
> I think the whole problem is that the udl interface has leaked into the sdk-common.

With udl definitions you mean the rust structs that are also defined in the udl? (asking because the udl file is still in sdk-bindings)

**Comment by erdemyerebasmaz** [comment]
Despite being more error-prone than your suggestions, the current approach of using feature flags—`#[cfg(feature = "liquid")]` and `#[cfg(not(feature = "liquid"))]`—to include or exclude fields from shared structs is imo adequate.

I am not sure if this is worth doubling the work over, we can add validation tests for `sdk-common` w/ & w/o `liquid` feature on the CI to catch any issues.

I'm open to any solutions that does not double the work managing these structs.

**Comment by roeierez** [comment]
> > I think the whole problem is that the udl interface has leaked into the sdk-common.
> 
> With udl definitions you mean the rust structs that are also defined in the udl? (asking because the udl file is still in sdk-bindings)

I am referring to the fact that the LnUrlPayRequest is defined in sdk common but is not used there anywhere, it is only used in sdk-core and liquid project. So in fact we can create for both of these referring projects their own LnUrlPayRequest. Not only that the LnUrlPayRequest is the interface implementation (udl file) so no reason both will share the same interface (or at least no reason to require that). Both interfaces have different motivation to change (different implementations).
